src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -39,3 +56,96 @@ def create_query_generator(plan: SnowflakePlan) -> QueryGenerator:
         )
 
     return QueryGenerator(plan.session, snowflake_create_table_plan_info)
+
+
+def replace_child(
+    parent: LogicalPlan,
+    old_child: LogicalPlan,
+    new_child: LogicalPlan,
+    analyzer: Analyzer,
+) -> None:
+    """
+    Helper function to replace the child node in the plan with a new child.
+
+    Whenever necessary, we convert the new_child into a Selectable or SnowflakePlan
+    based on the parent node type.
+    """
+
+    def to_selectable(plan: LogicalPlan, analyzer: Analyzer) -> Selectable:
+        """Given a LogicalPlan, convert it to a Selectable."""
+        if isinstance(plan, Selectable):
+            return plan
+
+        snowflake_plan = analyzer.resolve(plan)
+        return SelectSnowflakePlan(snowflake_plan, analyzer=analyzer)
```

Comment:
@sfc-gh-yzou does this look right to you for handling CTE related nodes

Reply:
yes, and i think this handles selecable correctly also.